### ISSUE TITLE
oga 0.3.0 not compatible with sax-machine 1.3.1

### ISSUE THREAD

**Issue by darix** [pauldix/sax-machine#69]
Reproducer:

```ruby
require 'feedjira';
require 'open-uri';
SAXMachine.handler = :oga
Feedjira::Feed.parse(open('http://feedjira.com/blog/feed.xml').read)

NoMethodError: undefined method `name' for ["xmlns", "http://www.w3.org/2005/Atom"]:Array
from /usr/lib64/ruby/gems/2.2.0/gems/sax-machine-1.3.1/lib/sax-machine/handlers/sax_oga_handler.rb:17:in `block in on_element'
```
yorick mentioned the the following links as part of the discussion:

```
18:03:52 < yorickpeterse> darix: https://github.com/YorickPeterse/oga/issues/92
18:06:44 < yorickpeterse> https://github.com/YorickPeterse/oga/commit/d8b9725b82f93d92b10170612446fbbef6190fda actual commit  
18:06:50 < yorickpeterse> basically attributes are now hashes in the SAX API
18:07:04 < yorickpeterse> Although you can override whatever it should be
18:15:39 < yorickpeterse> Basically in the past on_element's signature would be on_element(String, String, Array<Oga::XML::Attribute>) IIRC
18:15:47 < yorickpeterse> That has been changed to on_element(String, String, Hash) 
```

Switching to nokogiri works as work around for now.

**Comment by krasnoukhov** [comment]
Hello @darix, thanks for reporting. Well, Oga 0.3.0 introduced regression in decoding XML entities, so it's not ready to be used with with sax-machine. This fact is acknowleged by @YorickPeterse (see [comment](https://github.com/YorickPeterse/oga/issues/91#issuecomment-91005229)) so I'm pretty sure this will be fixed soon.

Meanwhile, there is [oga branch](https://github.com/pauldix/sax-machine/tree/oga) which works around the issue. You can use that or switch to nokogiri/ox as a handler while this is being in progress.

I'll close this issue once new version of Oga is released.

**Comment by krasnoukhov** [comment]
[1.3.2](https://github.com/pauldix/sax-machine/releases/tag/v1.3.2) is released to support Oga >= 0.3.4. Thanks for reporting!
